src/GogeDao.sol

```diff
@@ -370,718 +476,955 @@ contract GogeDAO is Ownable {
+        pollStartTime[pollNum] = block.timestamp;
+        pollEndTime[pollNum]   = _change.endTime;
+        pollAuthor[pollNum]    = msg.sender;
+
+        activePolls.push(pollNum);
+    }
+
+    /// @notice A method for a voter to add a vote to an existing poll.
     /// @param  _pollNum The poll number.
     /// @param  _numVotes The size of the vote to be created.
     function addVote(uint256 _pollNum, uint256 _numVotes) public {
+        require(_pollNum <= pollNum, "GogeDao.sol::addVote() Poll doesn't Exist");
+        require(block.timestamp >= pollStartTime[_pollNum] && block.timestamp < pollEndTime[_pollNum], "GogeDao.sol::addVote() Poll Closed");
+        require(isActivePoll(_pollNum), "GogeDao.sol::addVote() Poll is not active");
 
-        require(block.timestamp - IGogeERC20(governanceTokenAddr).getLastReceived(_msgSender()) >= (5 minutes), "Must wait 5 minutes after purchasing tokens to place any votes.");
-        require(_pollNum <= pollNum, "Poll doesn't Exist");
-        require(IGogeERC20(governanceTokenAddr).balanceOf(_msgSender()) >= _numVotes, "Exceeds Balance");
-        require(block.timestamp >= pollStartTime[_pollNum] && block.timestamp < pollEndTime[_pollNum], "Poll Closed");
-        require(IGogeERC20(governanceTokenAddr).transferFrom(_msgSender(), address(this), _numVotes));
+        require(block.timestamp - IGogeERC20(governanceTokenAddr).getLastReceived(msg.sender) >= (5 minutes), "GogeDao.sol::addVote() Must wait 5 minutes after purchasing tokens to place any votes.");
+        require(IGogeERC20(governanceTokenAddr).balanceOf(msg.sender) >= _numVotes, "GogeDao.sol::addVote() Exceeds Balance");
+        require(IGogeERC20(governanceTokenAddr).transferFrom(msg.sender, address(this), _numVotes));
 
-        voterLibrary[_pollNum].push(_msgSender());
-        polls[_pollNum][_msgSender()] = _numVotes;
-        totalVotes[_pollNum] += _numVotes;
-        historicalTally[_pollNum] += _numVotes;
+        _addToVoterLibrary(_pollNum, msg.sender);
+        _addToAdvocateFor(_pollNum, msg.sender);
 
-        bool quorumMet = ( totalVotes[_pollNum] * 100 / IGogeERC20(governanceTokenAddr).getCirculatingMinusReserve() ) >= quorum;
+        polls[_pollNum][msg.sender] += _numVotes;
+        totalVotes[_pollNum]        += _numVotes;
+
+        bool quorumMet = getProportion(_pollNum) >= quorum;
         bool enactChange = false;
 
-        if (!veto && quorumMet) {
-            enactChange = true;
-        }
-        else if (vetoAuthority[_msgSender()] && quorumMet) {
-            enactChange = true;
+        if (!gateKeeping && quorumMet) enactChange = true;
+        else if (gateKeeper[msg.sender] && quorumMet) enactChange = true;
+
+        if (enactChange) _executeProposal(_pollNum);
+    }
+
+    /// @notice A method for a voter to remove their votes from all active polls.
+    function removeAllVotes() public {
+        for (uint256 i = 0; i < activePolls.length; i++) {
+            _removeVote(activePolls[i]);
```

Comment:
Because removeAllVotes will remove all votes a person has an all activePolls. Could have votes on just 1 poll or many, but this will go through all activePolls and refund you for any activePoll you have votes for. _removeVote will remove the vote from a single poll at a time.